Document/0x04f-Testing-Network-Communication.md

```diff
@@ -205,67 +205,123 @@ If you want to verify whether your server supports the right cipher suites, ther
 
 To use the interception proxy, you'll need to run it on your host computer and configure the mobile app to route HTTP(S) requests to your proxy. In most cases, it is enough to set a system-wide proxy in the network settings of the mobile device - if the app uses standard HTTP APIs or popular libraries such as `okhttp`, it will automatically use the system settings.
 
 <img src="Images/Chapters/0x04f/BURP.png" width="100%" />
 
-Using a proxy breaks SSL certificate verification and the app will usually fail to initiate TLS connections. To work around this issue, you can install your proxy's CA certificate on the device. We'll explain how to do this in the OS-specific "Basic Security Testing" chapters.
+### Installing the Proxy Certificate
 
-## Intercepting Non-HTTP Traffic
+Using an interception proxy breaks SSL certificate verification and the app will usually fail to initiate TLS connections. So, interception proxies require you to install a custom CA certificate on the mobile device, which allows the proxy to decrypt and inspect the encrypted HTTPS traffic. However, some apps implement certificate pinning to prevent this, which requires additional steps to bypass.
+
+### Per-Platform Instructions
+
+- Android: see @MASTG-TECH-0011
+- iOS: see @MASTG-TECH-0063
+
+## MASTG-TECH: Intercepting Non-HTTP Traffic Using an Interception Proxy
 
 Interception proxies such as @MASTG-TOOL-0077 and @MASTG-TOOL-0079 won't show non-HTTP traffic, because they aren't capable of decoding it properly by default. There are, however, Burp plugins available such as:
 
 - [Burp-non-HTTP-Extension](https://github.com/summitt/Burp-Non-HTTP-Extension "Burp-non-HTTP-Extension") and
 - [Mitm-relay](https://github.com/jrmdev/mitm_relay "Mitm-relay").
 
-These plugins can visualize non-HTTP protocols and you will also be able to intercept and manipulate the traffic.
+These plugins can visualize non-HTTP protocols, allowing you to intercept and manipulate the traffic.
 
 Note that this setup can sometimes become very tedious and is not as straightforward as testing HTTP.
 
-## Intercepting Traffic from the App Process
+## MASTG-TECH: Intercepting HTTP Traffic by Hooking Network APIs at the Application Layer
 
 Depending on your goal while testing the app, sometimes it is enough to monitor the traffic before it reaches the network layer or when the responses are received in the app.
 
-You don't need to deploy a fully fledged MITM attack if you simply want to know if a certain piece of sensitive data is being transmitted to the network. In this case you wouldn't even have to bypass pinning, if implemented. You just have to hook the right functions, e.g. `SSL_write` and `SSL_read` from openssl.
+This means that you don't need to deploy a fully fledged MITM attack (including ARP Spoofing attacks, etc.) if you simply want to know if a certain piece of sensitive data is being transmitted to the network. In this case, you wouldn't even have to bypass pinning, if implemented.
```

Comment:
```suggestion
This means that you don't need to deploy a fully fledged MITM attack (including ARP Spoofing attacks, etc.) if you simply want to determine if certain sensitive data is being transmitted to the network. With this approach, you will not interfere with any TLS verification or pinning.
```
Can we maybe rephrase this section to a general 'Pro' and 'Con' ? And we can do that for each technique then. Or 'Advantages' and 'Limitations' ? The content itself is already there, but that gives us nicer consistency and the ability to quickly compare.